### PR TITLE
ieee802154: Expose configurations to Kconfig

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR exposes IEEE802.15.4 configurations to Kconfig

### Testing procedure
- Testing with e.g `default` and a IEEE802.15.4 radio (e.g at86rf233 on samr21-xpro).
- Change some settings (Channel, Power, etc) and check that they reflect on the binary (use `ifconfig` to check these settings)
- Check that it's still possible to configure the radios without Kconfig (e.g DEFAULT_CHANNEL or CFLAGS)

### Issues/PRs references
Part of #12888

### CODE REVIEW

File: sys/net/link_layer/Kconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2019 HAW Hamburg
```
Comment (jia200x) [inline sys/net/link_layer/Kconfig]: Oops, I forgot to update the year

File: sys/net/link_layer/ieee802154/Kconfig [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    help
+        Configure IEEE802.15.4 module using Kconfig
+
+if KCONFIG_MODULE_IEEE802154
+
+    config IEEE802154_DEFAULT_SUBGHZ_CHANNEL
+        int "IEEE802.15.4 default sub-GHZ channel"
+        default 5
+        range 0 11
+        help
```
Comment (leandrolanzieri) [inline sys/net/link_layer/ieee802154/Kconfig]: Here and below, we can drop the help string, as it is the same as the prompt.

File: sys/net/link_layer/ieee802154/Kconfig [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2019 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+menuconfig KCONFIG_MODULE_IEEE802154
+    bool "IEEE802.15.4 configurations"
```
Comment (leandrolanzieri) [inline sys/net/link_layer/ieee802154/Kconfig]: For consistency
```suggestion
    bool "Configure IEEE802.15.4"
```

File: sys/net/link_layer/ieee802154/Kconfig [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    depends on MODULE_IEEE802154
+    help
+        Configure IEEE802.15.4 module using Kconfig
+
+if KCONFIG_MODULE_IEEE802154
+
+    config IEEE802154_DEFAULT_SUBGHZ_CHANNEL
+        int "IEEE802.15.4 default sub-GHZ channel"
+        default 5
+        range 0 11
```
Comment (benpicco) [inline sys/net/link_layer/ieee802154/Kconfig]: This depends on the modulation and the frequency band. (see e.g. IEEE 802.15.4g-2012 Table 68d).
It's a mess - I'd just give up and drop the `range`.

File: sys/net/link_layer/ieee802154/Kconfig [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        help
+            IEEE802.15.4 default sub-GHZ channel
+            
+    config IEEE802154_DEFAULT_CHANNEL
+        int "IEEE802.15.4 default channel"
+        default 26
+        range 11 26
```
Comment (benpicco) [inline sys/net/link_layer/ieee802154/Kconfig]: same as for sub-GHz

image

**Comment by tcschmidt** [comment]
@jia200x @leandrolanzieri can you fix this?

**Comment by jia200x** [comment]
I pushed a fixup with the requested changes.

File: sys/net/link_layer/ieee802154/Kconfig (hunk 4 above)
Comment (jia200x) [reply sys/net/link_layer/ieee802154/Kconfig]: done!

File: sys/net/link_layer/ieee802154/Kconfig (hunk 5 above)
Comment (jia200x) [reply sys/net/link_layer/ieee802154/Kconfig]: done!

**Comment by leandrolanzieri** [comment]
Changes look good. Please squash.

When you rebase please make a grep to check that all macro names are updated, it seems that you will have to apply changes to the recently added `at86rf215.h` and `kw41zrf.h`.

**Comment by jia200x** [comment]
done! updated the `at86rf215` and `kw41zrf`

**Pull request merged**